src/main/java/software/sigma/internship/service/StudentServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package software.sigma.internship.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.sigma.internship.dao.repo.StudentRepository;
+import software.sigma.internship.user.Student;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class StudentServiceImpl implements StudentService {
+
+    private StudentRepository studentRepository;
+
+    @Override
+    public Student save(Student student) {
+        return studentRepository.save(student);
+    }
+
+    @Override
+    public List<Student> fetchList() {
+        return studentRepository.findAll();
+    }
+
+    @Override
+    public Student update(Student student, Long id) {
```

Comment:
Please read how studentRepository.save(student) works. I think it will create a new entity in case an entity doesn't exist or update when the entity already exists.

Reply:
It throws IllegalArgumentException – in case the given entity is null.